packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -435,8 +436,9 @@ function DashboardLayout(props: DashboardLayoutProps) {
 
   // If useEffect was used, the reset would also happen on the client render after SSR which we don't need
   React.useMemo(() => {
-    selectedItemIdRef.current = '';
-    // eslint-disable-next-line react-hooks/exhaustive-deps
+    if (navigation) {
```

Comment:
Having the selected item come from a context will eventually get rid of this so I guess this workaround should work for now.

Reply:
Have you seen that I added a bunch of helpers to optimize translating paths to navigation items and vice versa
https://github.com/mui/toolpad/blob/master/packages/toolpad-core/src/shared/navigation.tsx#L61-L166

it's being used by the new `PageContainer` [breadcrumbs](https://github.com/mui/toolpad/blob/master/packages/toolpad-core/src/useActivePage/useActivePage.ts) calculation. I expect a lot of it to be reusable for DashboardLayout. It adds some global caching to avoid recalculating these path => item maps over and over again.